test/optimizer/plugins/ConstructInlineInstanceOptimizer.test.ts

```diff
@@ -0,0 +1,550 @@
+
+        instance = new ExportableInstance('Foo');
+        instance.instanceOf = 'Patient';
+      });
+
+      it('should have appropriate metadata', () => {
+        expect(optimizer.name).toBe('construct_inline_instance');
+        expect(optimizer.description).toBeDefined();
+        expect(optimizer.runBefore).toBeUndefined();
+        expect(optimizer.runAfter).toBeUndefined();
+      });
+
+      it('should create inline instances from contained resources', () => {
+        instance.rules = [
+          containedResourceType1,
+          containedId1,
+          containedResourceType2,
+          containedId2
+        ];
+        const myPackage = new Package();
+        myPackage.add(instance);
+        optimizer.optimize(myPackage);
+
+        expect(myPackage.instances).toHaveLength(3);
+        assertExportableInstance(
+          myPackage.instances[1],
+          'Bar',
+          'Observation',
+          'Inline',
+          undefined,
+          undefined,
+          []
+        );
+        assertExportableInstance(
+          myPackage.instances[2],
+          'Baz',
+          'ValueSet',
+          'Inline',
+          undefined,
+          undefined,
+          []
+        );
+
+        const inlineInstanceRule1 = new ExportableAssignmentRule('contained[0]');
+        inlineInstanceRule1.value = 'Bar';
+        inlineInstanceRule1.isInstance = true;
+        const inlineInstanceRule2 = new ExportableAssignmentRule('contained[1]');
+        inlineInstanceRule2.value = 'Baz';
+        inlineInstanceRule2.isInstance = true;
+        expect(instance.rules).toEqual([inlineInstanceRule1, inlineInstanceRule2]);
+      });
+
+      it('should create inline instances from contained and bundle resources', () => {
+        instance.rules = [containedResourceType1, containedId1, bundleResourceType, bundleId];
+        const myPackage = new Package();
+        myPackage.add(instance);
+        optimizer.optimize(myPackage);
+
+        expect(myPackage.instances).toHaveLength(3);
+        assertExportableInstance(
+          myPackage.instances[1],
+          'Bar',
+          'Observation',
+          'Inline',
+          undefined,
+          undefined,
+          []
+        );
+        assertExportableInstance(
+          myPackage.instances[2],
+          'Bam',
+          'Organization',
+          'Inline',
+          undefined,
+          undefined,
+          []
+        );
+
+        const inlineInstanceRule1 = new ExportableAssignmentRule('contained[0]');
+        inlineInstanceRule1.value = 'Bar';
+        inlineInstanceRule1.isInstance = true;
+        const inlineInstanceRule2 = new ExportableAssignmentRule('entry[0].resource');
+        inlineInstanceRule2.value = 'Bam';
+        inlineInstanceRule2.isInstance = true;
+        expect(instance.rules).toEqual([inlineInstanceRule1, inlineInstanceRule2]);
+      });
+
+      it('should create an inline instance from a contained resource with non-numeric paths', () => {
```

Comment:
Again, if you want to be super defensive against any possible changes to our code in the future, then... OK.  But otherwise I don't think we currently generate paths like this.